building with lotus API

longfeiWan9 opened the pull request:
Closing filecoin-project/lotus-docs#23
Create a tutorial to teach developers how to build using lotus node, including:

-  Setting up local lotus node for Calibration TestNet
- Using filecon.js to write code to interact with Filecoin Network to
  - query basic blockchain info
  - manage lotus wallet
  - manage light wallet
  - make a storage deal

filecorgi commented:
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/build/hosted-lotus-node.md`:
- The following link failed to connect:
  http://localhost:8080/about-filecoin/how-filecoin-works/
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ake sure everything is set up properly. Especially ensure you have a local l...
                                             ^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...o back and do it following the guild to Install and launch a Lotus node  (op...
                                             ^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus source code for the latest release  Build and install lotus for Calibnet...
                                             ^^^^^
  ```
- A determiner may be missing.<!--CategoryID:GRAMMAR, RuleID: THE_SUPERLATIVE-->
  ```
  ...all lotus for Calibnet  Fast sync using latest snapshot (still need several ...
                                             ^^^^^^
  ```
- This phrase is duplicated. You should probably use “Lotus Wallet” only once.<!--CategoryID:GRAMMAR, RuleID: PHRASE_REPETITION-->
  ```
  ...ny node and represented by key pairs. # Lotus Wallet Lotus wallet is created...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- Consider using an alternative to ‘right’ for your writing stand out and sound more polished.<!--CategoryID:STYLE, RuleID: RIGHT_STYLE-->
  ```
  ... command line or RPC API calls with the right level of auth-token. First, le...
                                             ^^^^^
  ```
- Use a comma before “and” if it connects two independent clauses (unless they are closely connected and short).<!--CategoryID:PUNCTUATION, RuleID: COMMA_COMPOUND_SENTENCE_2-->
  ```
  ...ke sure your local lotus node is running and the wallet address is filled wi...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... token with sign permission via command line which you will use later in the...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ting a light wallet from mnemonic and a password which does not require you ...
                                             ^^^^^^^^
  ```
- The preposition ‘as’ seems more likely in this position.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_REPLACE_FOR_AS-->
  ```
  ...u to run a full lotus node the same way for a lotus wallet. As long as you h...
                                             ^^^
  ```
- As an alternative to the over-used intensifier ‘very’, consider replacing this phrase.<!--CategoryID:STYLE, RuleID: EN_WEAK_ADJECTIVE-->
  ```
  ... wallet using the BIP39 standard, it is very important to back up the mnemon...
                                             ^^^^^^^^^^^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... a wallet on the lotus node via RPC API call as well as create a light walle...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... decentralized storage network, storing data is definitely one of the most i...
                                             ^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.<!--CategoryID:STYLE, RuleID: EN_REPEATEDWORDS_IMPORTANT-->
  ```
  ...e. Let's dive into it. The operation of important data into the lotus node r...
                                             ^^^^^^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ows which file will be stored using its CID which is the unique fingerprint ...
                                             ^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...hich file you want to store by its CID. Then in order to store this data on ...
                                             ^^^^
  ```
- Consider a shorter alternative to avoid wordiness.<!--CategoryID:REDUNDANCY, RuleID: IN_ORDER_TO_PREMIUM-->
  ```
  ...file you want to store by its CID. Then in order to store this data on the F...
                                             ^^^^^^^^^^^
  ```
- File types are normally capitalized.<!--CategoryID:CASING, RuleID: FILE_EXTENSIONS_CASE-->
  ```
  ...ng storage deals with it. Let's run the js code to store a file on Filecoin....
                                             ^^
  ```
- Consider using “many”.<!--CategoryID:REDUNDANCY, RuleID: NUMEROUS_DIFFERENT-->
  ```
  ...he storage deal will need to go through many different processes to be final...
                                             ^^^^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus command line using its CID. See the Processing states  (opens new window...
                                             ^^^^^^^^^^
  ```

mishmosh commented:
@jenksguo @developerally Can you also follow this tutorial and add review comments sometime this week? I think your relative beginners' mindset would be very helpful here to ensure it's complete.

johnnymatthews commented on docs/.vuepress/config.js:
```diff
@@ -92,6 +92,7 @@ module.exports = {
 
           '/build/': [
             'get-started',
+            'hosted-lotus-node',
```
By _hosted_ do you mean running a Lotus node on a remote server somewhere?

longfeiWan9 replied:
You can either run this lotus node on your local machine or a remote server. It will server the same purpose

johnnymatthews commented on docs/build/hosted-lotus-node.md:
```diff
@@ -0,0 +1,643 @@
+
+Developers can also use the API client libraries to utilize the features of Filecoin Network since those libraries take care of the low-level logic of connecting to lotus node, requesting and handing the response via simple calls, etc. Currently, there are many libraries available:
+
++ [filecoin.js](https://filecoin-shipyard.github.io/filecoin.js/)
++ [filecoin-js-signer](https://github.com/blitslabs/filecoin-js-signer)
++ [Filecoin Signing Tools](https://github.com/Zondax/filecoin-signing-tools)
```
```suggestion
- [filecoin.js](https://filecoin-shipyard.github.io/filecoin.js/)
- [filecoin-js-signer](https://github.com/blitslabs/filecoin-js-signer)
- [Filecoin Signing Tools](https://github.com/Zondax/filecoin-signing-tools)
```

Use dashes `-` for list items.

johnnymatthews commented on docs/build/hosted-lotus-node.md:
```diff
@@ -0,0 +1,643 @@
++ [filecoin-js-signer](https://github.com/blitslabs/filecoin-js-signer)
++ [Filecoin Signing Tools](https://github.com/Zondax/filecoin-signing-tools)
+
+In this tutorial, we will run a  **local lotus node** and use **filecoin.js** library to build examples to walk you through: 
```
Bolding should be reserved for when the user/reader has to look for something, or click on something. https://docs.filecoin.io/community/contribute/grammar-formatting-and-style/#bold-text

johnnymatthews commented on docs/build/hosted-lotus-node.md:
```diff
@@ -0,0 +1,643 @@
+   - manage light wallet
+   - make a storage deal
+
+## **Pre-requirements**
```
Don't bold headers. https://docs.filecoin.io/community/contribute/grammar-formatting-and-style/#titles

johnnymatthews commented on docs/build/hosted-lotus-node.md:
```diff
@@ -0,0 +1,643 @@
++ [How to install a lotus node for the Calibration network](https://lotu.sh/lotus/install/prerequisites/#supported-platforms)
++ [How to use Lotus command line](https://lotu.sh/lotus/manage/lotus-cli/)
+
+Also, make sure you have all the dependencies installed:
+
++ Fully Synced Calibration TestNet lotus node
++ node.js
++ npm
```
This should go under a **Prerequisites** section.

johnnymatthews commented on docs/build/hosted-lotus-node.md:
```diff
@@ -0,0 +1,643 @@
+
+To sum up, you need to:
+
+1. **Download lotus source code for the latest release**
```
Don't bold list items. https://docs.filecoin.io/community/contribute/grammar-formatting-and-style/#bold-text

johnnymatthews commented on docs/build/hosted-lotus-node.md:
````diff
@@ -0,0 +1,643 @@
+
+4. **[Enable remote API access to your lotus node](https://lotu.sh/developers/api-access/)**
+
+If everything set up correctly, you should be able to check the status of your local node and wallet balance.
+
+```shell with-output
+#Check if the node sync to the highest tipset
+lotus sync wait
+```
+
+```
+Worker: 10161; Base: 778820; Target: 778820 (diff: 0)
+State: complete; Current Epoch: 778820; Todo: 0
+
+Done!
+```
+
+```shell with-output
+#Check wallet balance
+lotus wallet list 
+```
+
+```
+Address                                    Balance        Nonce  
+t154xvuihhicgluafwmohwzwmtmqp44pwwgewyvma  100 FIL          0
+```
````
Add these as individual steps that the user should follow through:

```
1. To make sure that everything is working correctly, check the status of your local node: [...]
1. Check your wallet balance: [...]
```

johnnymatthews commented on docs/build/hosted-lotus-node.md:
````diff
@@ -0,0 +1,643 @@
+   npm install filecoin.js
+   ```
+
+3. **Test a simple function call**
+
+   At this step, you are ready to write code to connect to your lotus node and interact with the lotus APIs. Let's write a simple script to query the lotus version to verify if everything works as expected.
+
+   First, let's create a `chainDataQuery.mjs` file in your project and add the following codes. We use `mjs` so we can use import statement in our code. 
+
+   ```javascript
+   //1. Import necessary packages
+   import { HttpJsonRpcConnector, LotusClient } from "filecoin.js";
+   
+   //2. Use the local node URL to create a connection to your lotus node
+   const localNodeUrl = "http://127.0.0.1:1234/rpc/v0";
+   const localConnector = new HttpJsonRpcConnector({ url: localNodeUrl });
+   
+   //3. lotusClient exposes all lotus APIs
+   const lotusClient = new LotusClient(localConnector);
+   const version = await lotusClient.common.version();
+   console.log(version);
+   ```
+
+   Run the following command in your project directory to test it.
+
+   ```shell with-output
+   ~/build-with-lotus $ node basicQuery.mjs
+   ```
+
+   ```shell
+   {
+     Version: '1.15.0+calibnet+git.0ac1bbc7a',
+     APIVersion: 66816,
+     BlockDelay: 30
+   }
+   ```
+
+   If everything is set up correctly, our code will be able to connect to the lotus node and request chain data back as shown above. And we are ready to explore more features that the lotus provides via API calls. 
````
Make this a subsection of **Create a node.js project**:

```
#### Test a simple function call

At this step, you are ready to write code to connect to your lotus node and interact with the lotus APIs. Let's write a simple script to query the lotus version to verify if everything works as expected.

1. First, let's create a `chainDataQuery.mjs` file in your project and add the following codes. We use `mjs` so we can use import statement in our code:

...
```

johnnymatthews commented on docs/build/hosted-lotus-node.md:
```diff
@@ -0,0 +1,643 @@
+
+As one of the API client libraries, filecion.js wrapped all the lotus APIs into simple method calls to provide  a complete library to interact with any lotus node. Let's look at some common packages of LotusClient so you can have a sense of what you can use. 
+
++ **LotusClient.common** - methods to get node info.
++ **LotusClient.chain** -  methods to interact with the blockchain.
++ **LotusClient.wallet** - methods to manage wallet on the lotus node.
++ **LotusClient.client** - methods to interact with the storage and retrieval markets as a client.
++ **LotusClient.paych** - methods to interact with and manage payment channels.
++ **LotusClient.state** - methods to query, inspect, and interact with chain state.
++ more ...
```
If these are variables of methods, they should be between code tags:

```
`LotusClient.common`
```

https://docs.filecoin.io/community/contribute/grammar-formatting-and-style/#inline-code-tags

filecorgi commented:
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/build/hosted-lotus-node.md`:
- The following link failed to connect:
  http://localhost:8080/about-filecoin/how-filecoin-works/
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...o back and do it following the guild to Install and launch a Lotus node  (op...
                                             ^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus source code for the latest release  Build and install lotus for Calibnet...
                                             ^^^^^
  ```
- A determiner may be missing.<!--CategoryID:GRAMMAR, RuleID: THE_SUPERLATIVE-->
  ```
  ...all lotus for Calibnet  Fast sync using latest snapshot (still need several ...
                                             ^^^^^^
  ```
- This phrase is duplicated. You should probably use “Lotus Wallet” only once.<!--CategoryID:GRAMMAR, RuleID: PHRASE_REPETITION-->
  ```
  ...ny node and represented by key pairs. # Lotus Wallet Lotus wallet is created...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- Use a comma before “and” if it connects two independent clauses (unless they are closely connected and short).<!--CategoryID:PUNCTUATION, RuleID: COMMA_COMPOUND_SENTENCE_2-->
  ```
  ...ke sure your local lotus node is running and the wallet address is filled wi...
                                             ^^^^
  ```
- As an alternative to the over-used intensifier ‘very’, consider replacing this phrase.<!--CategoryID:STYLE, RuleID: EN_WEAK_ADJECTIVE-->
  ```
  ... wallet using the BIP39 standard, it is very important to back up the mnemon...
                                             ^^^^^^^^^^^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.<!--CategoryID:STYLE, RuleID: EN_REPEATEDWORDS_IMPORTANT-->
  ```
  ...e. Let's dive into it. The operation of important data into the lotus node r...
                                             ^^^^^^^^^
  ```
- Consider a shorter alternative to avoid wordiness.<!--CategoryID:REDUNDANCY, RuleID: IN_ORDER_TO_PREMIUM-->
  ```
  ...file you want to store by its CID. Then in order to store this data on the F...
                                             ^^^^^^^^^^^
  ```
- File types are normally capitalized.<!--CategoryID:CASING, RuleID: FILE_EXTENSIONS_CASE-->
  ```
  ...ng storage deals with it. Let's run the js code to store a file on Filecoin....
                                             ^^
  ```
- Consider using “many”.<!--CategoryID:REDUNDANCY, RuleID: NUMEROUS_DIFFERENT-->
  ```
  ...he storage deal will need to go through many different processes to be final...
                                             ^^^^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus command line using its CID. See the Processing states  (opens new window...
                                             ^^^^^^^^^^
  ```

filecorgi commented:
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/build/hosted-lotus-node.md`:
- The following link failed to connect:
  http://localhost:8080/about-filecoin/how-filecoin-works/
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...o back and do it following the guild to Install and launch a Lotus node  (op...
                                             ^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus source code for the latest release  Build and install Lotus for Calibnet...
                                             ^^^^^
  ```
- A determiner may be missing.<!--CategoryID:GRAMMAR, RuleID: THE_SUPERLATIVE-->
  ```
  ...all Lotus for Calibnet  Fast sync using latest snapshot (still need several ...
                                             ^^^^^^
  ```
- This phrase is duplicated. You should probably use “Lotus Wallet” only once.<!--CategoryID:GRAMMAR, RuleID: PHRASE_REPETITION-->
  ```
  ...ny node and represented by key pairs. # Lotus Wallet Lotus wallet is created...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- Use a comma before “and” if it connects two independent clauses (unless they are closely connected and short).<!--CategoryID:PUNCTUATION, RuleID: COMMA_COMPOUND_SENTENCE_2-->
  ```
  ...ke sure your local Lotus node is running and the wallet address is filled wi...
                                             ^^^^
  ```
- As an alternative to the over-used intensifier ‘very’, consider replacing this phrase.<!--CategoryID:STYLE, RuleID: EN_WEAK_ADJECTIVE-->
  ```
  ... wallet using the BIP39 standard, it is very important to back up the mnemon...
                                             ^^^^^^^^^^^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.<!--CategoryID:STYLE, RuleID: EN_REPEATEDWORDS_IMPORTANT-->
  ```
  ...e. Let's dive into it. The operation of important data into the Lotus node r...
                                             ^^^^^^^^^
  ```
- Consider a shorter alternative to avoid wordiness.<!--CategoryID:REDUNDANCY, RuleID: IN_ORDER_TO_PREMIUM-->
  ```
  ...file you want to store by its CID. Then in order to store this data on the F...
                                             ^^^^^^^^^^^
  ```
- File types are normally capitalized.<!--CategoryID:CASING, RuleID: FILE_EXTENSIONS_CASE-->
  ```
  ...ng storage deals with it. Let's run the js code to store a file on Filecoin....
                                             ^^
  ```
- Consider using “many”.<!--CategoryID:REDUNDANCY, RuleID: NUMEROUS_DIFFERENT-->
  ```
  ...he storage deal will need to go through many different processes to be final...
                                             ^^^^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus command line using its CID. See the Processing states  (opens new window...
                                             ^^^^^^^^^^
  ```

longfeiWan9 commented:
> This looks like a great guide so far! Left a few comment for changes though, so I'll convert this PR to a draft since there's still a bit of work to be done before we can merge it.

@johnnymatthews  I have updated this PR according to your comments. Please take a look again. thanks.

mishmosh commented on docs/build/hosted-lotus-node.md:
```diff
@@ -0,0 +1,647 @@
+
+### Test a simple function call
+
+At this step, you are ready to write code to connect to your Lotus node and interact with the Lotus APIs. Let's write a simple script to query the Lotus version to verify if everything works as expected.
```
```suggestion
Now you're ready to write code to connect to your Lotus node and interact with the Lotus APIs. Let's write a simple script to query the Lotus version to verify if everything works as expected.
```

mishmosh commented on docs/build/hosted-lotus-node.md:
````diff
@@ -0,0 +1,647 @@
+    console.log(version);
+    ```
+
+2. Run the following command in your project directory to test it.
````
```suggestion
2. Run the following command in your project directory:
```

mishmosh commented on docs/build/hosted-lotus-node.md:
````diff
@@ -0,0 +1,647 @@
+    }
+    ```
+
+If everything is set up correctly, our code will be able to connect to the Lotus node and request chain data back as shown above. And we are ready to explore more features that the Lotus provides via API calls. 
````
```suggestion
If everything is set up correctly, your code will be able to connect to the Lotus node and return the chain data as shown above. Now, you're ready to explore more API features.
```

mishmosh commented on docs/build/hosted-lotus-node.md:
```diff
@@ -0,0 +1,647 @@
+As one of the API client libraries, filecion.js wrapped all the Lotus APIs into simple method calls to provide  a complete library to interact with any Lotus node. Let's look at some common packages of LotusClient so you can have a sense of what you can use. 
+
+- `LotusClient.common` - methods to get node info.
+- `LotusClient.chain` -  methods to interact with the blockchain.
```
```suggestion
- `LotusClient.chain` - methods to interact with the blockchain.
```
remove extra space

mishmosh commented on docs/build/hosted-lotus-node.md:
```diff
@@ -0,0 +1,647 @@
+
+As one of the API client libraries, filecion.js wrapped all the Lotus APIs into simple method calls to provide  a complete library to interact with any Lotus node. Let's look at some common packages of LotusClient so you can have a sense of what you can use. 
+
+- `LotusClient.common` - methods to get node info.
```
remove all the periods at the end of lines 165-171 (in lists, skip the periods if it's not a complete sentence)

mishmosh reviewed (changes requested):
I've only gotten partway through this review, but wanted to send it over so you can work on these comments in parallel.

filecorgi commented:
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/build/hosted-lotus-node.md`:
- The following link failed to connect:
  http://localhost:8080/about-filecoin/how-filecoin-works/
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ake sure everything is set up properly. Especially ensure you have a local L...
                                             ^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...o back and do it following the guild to Install and launch a Lotus node  (op...
                                             ^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus source code for the latest release  Build and install Lotus for Calibnet...
                                             ^^^^^
  ```
- A determiner may be missing.<!--CategoryID:GRAMMAR, RuleID: THE_SUPERLATIVE-->
  ```
  ...all Lotus for Calibnet  Fast sync using latest snapshot (still need several ...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ... - methods to get node info LotusClient.chain - methods to interact with the...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...nteract with the blockchain LotusClient.wallet - methods to manage wallet on...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...ge wallet on the Lotus node LotusClient.client - methods to interact with th...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...and manage payment channels LotusClient.state - methods to query, inspect, a...
                                             ^^^^^
  ```
- This phrase is duplicated. You should probably use “Lotus Wallet” only once.<!--CategoryID:GRAMMAR, RuleID: PHRASE_REPETITION-->
  ```
  ...ny node and represented by key pairs. # Lotus Wallet Lotus wallet is created...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- Use a comma before “and” if it connects two independent clauses (unless they are closely connected and short).<!--CategoryID:PUNCTUATION, RuleID: COMMA_COMPOUND_SENTENCE_2-->
  ```
  ...ke sure your local Lotus node is running and the wallet address is filled wi...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... token with sign permission via command line which you will use later in the...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ting a light wallet from mnemonic and a password which does not require you ...
                                             ^^^^^^^^
  ```
- The preposition ‘as’ seems more likely in this position.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_REPLACE_FOR_AS-->
  ```
  ...u to run a full Lotus node the same way for a Lotus wallet. As long as you h...
                                             ^^^
  ```
- As an alternative to the over-used intensifier ‘very’, consider replacing this phrase.<!--CategoryID:STYLE, RuleID: EN_WEAK_ADJECTIVE-->
  ```
  ... wallet using the BIP39 standard, it is very important to back up the mnemon...
                                             ^^^^^^^^^^^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... a wallet on the Lotus node via RPC API call as well as create a light walle...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... decentralized storage network, storing data is definitely one of the most i...
                                             ^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.<!--CategoryID:STYLE, RuleID: EN_REPEATEDWORDS_IMPORTANT-->
  ```
  ...e. Let's dive into it. The operation of important data into the Lotus node r...
                                             ^^^^^^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ows which file will be stored using its CID which is the unique fingerprint ...
                                             ^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...hich file you want to store by its CID. Then in order to store this data on ...
                                             ^^^^
  ```
- Consider a shorter alternative to avoid wordiness.<!--CategoryID:REDUNDANCY, RuleID: IN_ORDER_TO_PREMIUM-->
  ```
  ...file you want to store by its CID. Then in order to store this data on the F...
                                             ^^^^^^^^^^^
  ```
- File types are normally capitalized.<!--CategoryID:CASING, RuleID: FILE_EXTENSIONS_CASE-->
  ```
  ...ng storage deals with it. Let's run the js code to store a file on Filecoin....
                                             ^^
  ```
- Consider using “many”.<!--CategoryID:REDUNDANCY, RuleID: NUMEROUS_DIFFERENT-->
  ```
  ...he storage deal will need to go through many different processes to be final...
                                             ^^^^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus command line using its CID. See the Processing states  (opens new window...
                                             ^^^^^^^^^^
  ```

filecorgi commented:
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/build/building-with-lotus-APIs.md`:
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ake sure everything is set up properly. Especially ensure you have a local L...
                                             ^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...o back and do it following the guild to Install and launch a Lotus node  (op...
                                             ^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus source code for the latest release  Build and install Lotus for Calibnet...
                                             ^^^^^
  ```
- A determiner may be missing.<!--CategoryID:GRAMMAR, RuleID: THE_SUPERLATIVE-->
  ```
  ...all Lotus for Calibnet  Fast sync using latest snapshot (still need several ...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ... - methods to get node info LotusClient.chain - methods to interact with the...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...nteract with the blockchain LotusClient.wallet - methods to manage wallet on...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...ge wallet on the Lotus node LotusClient.client - methods to interact with th...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...and manage payment channels LotusClient.state - methods to query, inspect, a...
                                             ^^^^^
  ```
- This phrase is duplicated. You should probably use “Lotus Wallet” only once.<!--CategoryID:GRAMMAR, RuleID: PHRASE_REPETITION-->
  ```
  ...ny node and represented by key pairs. # Lotus Wallet Lotus wallet is created...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- Use a comma before “and” if it connects two independent clauses (unless they are closely connected and short).<!--CategoryID:PUNCTUATION, RuleID: COMMA_COMPOUND_SENTENCE_2-->
  ```
  ...ke sure your local Lotus node is running and the wallet address is filled wi...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... token with sign permission via command line which you will use later in the...
                                             ^^^^
  ```
- As an alternative to the over-used intensifier ‘very’, consider replacing this phrase.<!--CategoryID:STYLE, RuleID: EN_WEAK_ADJECTIVE-->
  ```
  ... wallet using the BIP39 standard, it is very important to back up the mnemon...
                                             ^^^^^^^^^^^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... a wallet on the Lotus node via RPC API call as well as create a light walle...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... decentralized storage network, storing data is definitely one of the most i...
                                             ^^^^
  ```
- This word has been used in one of the immediately preceding sentences. Using a synonym could make your text more interesting to read, unless the repetition is intentional.<!--CategoryID:STYLE, RuleID: EN_REPEATEDWORDS_IMPORTANT-->
  ```
  ...e. Let's dive into it. The operation of important data into the Lotus node r...
                                             ^^^^^^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ows which file will be stored using its CID which is the unique fingerprint ...
                                             ^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...hich file you want to store by its CID. Then in order to store this data on ...
                                             ^^^^
  ```
- Consider a shorter alternative to avoid wordiness.<!--CategoryID:REDUNDANCY, RuleID: IN_ORDER_TO_PREMIUM-->
  ```
  ...file you want to store by its CID. Then in order to store this data on the F...
                                             ^^^^^^^^^^^
  ```
- File types are normally capitalized.<!--CategoryID:CASING, RuleID: FILE_EXTENSIONS_CASE-->
  ```
  ...ng storage deals with it. Let's run the js code to store a file on Filecoin....
                                             ^^
  ```
- Consider using “many”.<!--CategoryID:REDUNDANCY, RuleID: NUMEROUS_DIFFERENT-->
  ```
  ...he storage deal will need to go through many different processes to be final...
                                             ^^^^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus command line using its CID. See the Processing states  (opens new window...
                                             ^^^^^^^^^^
  ```

filecorgi commented:
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/build/building-with-lotus-APIs.md`:
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ake sure everything is set up properly. Especially ensure you have a local L...
                                             ^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...o back and do it following the guild to Install and launch a Lotus node  (op...
                                             ^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus source code for the latest release  Build and install Lotus for Calibnet...
                                             ^^^^^
  ```
- A determiner may be missing.<!--CategoryID:GRAMMAR, RuleID: THE_SUPERLATIVE-->
  ```
  ...all Lotus for Calibnet  Fast sync using latest snapshot (still need several ...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ... - methods to get node info LotusClient.chain - methods to interact with the...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...nteract with the blockchain LotusClient.wallet - methods to manage wallet on...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...ge wallet on the Lotus node LotusClient.client - methods to interact with th...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...and manage payment channels LotusClient.state - methods to query, inspect, a...
                                             ^^^^^
  ```
- This phrase is duplicated. You should probably use “Lotus Wallet” only once.<!--CategoryID:GRAMMAR, RuleID: PHRASE_REPETITION-->
  ```
  ...ny node and represented by key pairs. # Lotus Wallet Lotus wallet is created...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- Consider using an alternative word for ‘right’ so that your writing stands out and sounds more polished.<!--CategoryID:STYLE, RuleID: RIGHT_STYLE-->
  ```
  ... command line or RPC API calls with the right level of auth-token. First, le...
                                             ^^^^^
  ```
- Use a comma before “and” if it connects two independent clauses (unless they are closely connected and short).<!--CategoryID:PUNCTUATION, RuleID: COMMA_COMPOUND_SENTENCE_2-->
  ```
  ...ke sure your local Lotus node is running and the wallet address is filled wi...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... token with sign permission via command line which you will use later in the...
                                             ^^^^
  ```
- As an alternative to the over-used intensifier ‘very’, consider replacing this phrase.<!--CategoryID:STYLE, RuleID: EN_WEAK_ADJECTIVE-->
  ```
  ... wallet using the BIP39 standard, it is very important to back up the mnemon...
                                             ^^^^^^^^^^^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... a wallet on the Lotus node via RPC API call as well as create a light walle...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... decentralized storage network, storing data is definitely one of the most i...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ows which file will be stored using its CID which is the unique fingerprint ...
                                             ^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...hich file you want to store by its CID. Then in order to store this data on ...
                                             ^^^^
  ```
- Consider a shorter alternative to avoid wordiness.<!--CategoryID:REDUNDANCY, RuleID: IN_ORDER_TO_PREMIUM-->
  ```
  ...file you want to store by its CID. Then in order to store this data on the F...
                                             ^^^^^^^^^^^
  ```
- File types are normally capitalized.<!--CategoryID:CASING, RuleID: FILE_EXTENSIONS_CASE-->
  ```
  ...ng storage deals with it. Let's run the js code to store a file on Filecoin....
                                             ^^
  ```
- Consider using “many”.<!--CategoryID:REDUNDANCY, RuleID: NUMEROUS_DIFFERENT-->
  ```
  ...he storage deal will need to go through many different processes to be final...
                                             ^^^^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus command line using its CID. See the Processing states  (opens new window...
                                             ^^^^^^^^^^
  ```

filecorgi commented:
- Image optimization came back clean!
- Vuepress build was successful!
### `docs/build/building-with-lotus-APIs.md`:
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ake sure everything is set up properly. Especially ensure you have a local L...
                                             ^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...o back and do it following the guild to Install and launch a Lotus node  (op...
                                             ^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus source code for the latest release  Build and install Lotus for Calibnet...
                                             ^^^^^
  ```
- A determiner may be missing.<!--CategoryID:GRAMMAR, RuleID: THE_SUPERLATIVE-->
  ```
  ...all Lotus for Calibnet  Fast sync using latest snapshot (still need several ...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ... - methods to get node info LotusClient.chain - methods to interact with the...
                                             ^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...nteract with the blockchain LotusClient.wallet - methods to manage wallet on...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...ge wallet on the Lotus node LotusClient.client - methods to interact with th...
                                             ^^^^^^
  ```
- If a new sentence starts here, add a space and start with an uppercase letter.<!--CategoryID:CASING, RuleID: LC_AFTER_PERIOD-->
  ```
  ...and manage payment channels LotusClient.state - methods to query, inspect, a...
                                             ^^^^^
  ```
- This phrase is duplicated. You should probably use “Lotus Wallet” only once.<!--CategoryID:GRAMMAR, RuleID: PHRASE_REPETITION-->
  ```
  ...ny node and represented by key pairs. # Lotus Wallet Lotus wallet is created...
                                             ^^^^^^^^^^^^^^^^^^^^^^^^^
  ```
- Consider using an alternative word for ‘right’ so that your writing stands out and sounds more polished.<!--CategoryID:STYLE, RuleID: RIGHT_STYLE-->
  ```
  ... command line or RPC API calls with the right level of auth-token. First, le...
                                             ^^^^^
  ```
- Use a comma before “and” if it connects two independent clauses (unless they are closely connected and short).<!--CategoryID:PUNCTUATION, RuleID: COMMA_COMPOUND_SENTENCE_2-->
  ```
  ...ke sure your local Lotus node is running and the wallet address is filled wi...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... token with sign permission via command line which you will use later in the...
                                             ^^^^
  ```
- As an alternative to the over-used intensifier ‘very’, consider replacing this phrase.<!--CategoryID:STYLE, RuleID: EN_WEAK_ADJECTIVE-->
  ```
  ... wallet using the BIP39 standard, it is very important to back up the mnemon...
                                             ^^^^^^^^^^^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... a wallet on the Lotus node via RPC API call as well as create a light walle...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ... decentralized storage network, storing data is definitely one of the most i...
                                             ^^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...ows which file will be stored using its CID which is the unique fingerprint ...
                                             ^^^
  ```
- Possible missing comma found.<!--CategoryID:MISC, RuleID: AI_HYDRA_LEO_MISSING_COMMA-->
  ```
  ...hich file you want to store by its CID. Then in order to store this data on ...
                                             ^^^^
  ```
- Consider a shorter alternative to avoid wordiness.<!--CategoryID:REDUNDANCY, RuleID: IN_ORDER_TO_PREMIUM-->
  ```
  ...file you want to store by its CID. Then in order to store this data on the F...
                                             ^^^^^^^^^^^
  ```
- File types are normally capitalized.<!--CategoryID:CASING, RuleID: FILE_EXTENSIONS_CASE-->
  ```
  ...ng storage deals with it. Let's run the js code to store a file on Filecoin....
                                             ^^
  ```
- Consider using “many”.<!--CategoryID:REDUNDANCY, RuleID: NUMEROUS_DIFFERENT-->
  ```
  ...he storage deal will need to go through many different processes to be final...
                                             ^^^^^^^^^^^^^^
  ```
- Only proper nouns start with an uppercase character (there are exceptions for headlines).<!--CategoryID:CASING, RuleID: EN_UPPER_CASE_NGRAM-->
  ```
  ...tus command line using its CID. See the Processing states  (opens new window...
                                             ^^^^^^^^^^
  ```

longfeiWan9 commented:
Hi, @johnnymatthews and @mishmosh. I have updated the tutorial according to your suggestions. Could you please take another look? Thank you.

mishmosh commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+---
+title: Building with Lotus APIs
+description: Lotus node provides comprehensive features via its JSON-RPC APIs for developers and builders to build on the Filecoin network. Developers can use any API client library to connect to Lotus node to use the build-in Lotus APIs.
```
```suggestion
description: Lotus offers a built-in JSON-RPC API for developers who need direct, protocol-level control for building tools and applications. Several client libraries are available.
```

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+weight: 40
+---
+
+In the Filecoin ecosystem, there are various node solutions to connect to the Filecoin network and provide  functions via RPC API calls. Developers can either install and run their own Lotus node locally to interact with the Filecoin network, or connect to other Lotus nodes hosted by Glif or Infura.
```
```suggestion
In the Filecoin ecosystem, there are various node solutions to connect to the Filecoin network and provide functions via RPC API calls. Developers can either install and run their own Lotus node locally to interact with the Filecoin network, or connect to other Lotus nodes hosted by Glif or Infura.
```

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+- [How to install a Lotus node for the Calibration network](https://lotu.sh/lotus/install/prerequisites/#supported-platforms)
+- [How to use Lotus command line](https://lotu.sh/lotus/manage/lotus-cli/)
+
+Also, make sure you have all the dependencies installed:
+
+- Fully Synced Calibration TestNet Lotus node
+- node.js
+- npm
```
```suggestion
Also, make sure you have all the dependencies installed:

- A fully-synced Lotus node on the calibration testnet
- Node.js
- NPM
```

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+### Set up lotus node
+
+If you have not installed a Lotus node locally and fully synced for Calibration TestNet, you should go back and do it following the guild to [Install and launch a Lotus node](https://lotus.filecoin.io/docs/set-up/install/).
+
+To sum up, you need to:
```
```suggestion
```

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+
+To sum up, you need to:
+
+1. Download Lotus source code for the latest release
```
```suggestion
1. Download Lotus source code for the latest release:
```

johnnymatthews commented:
There's quite a few changes needed here, but it'll just be faster for myself to make them since this PR has been building for a while. I do this now.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+2. Build and install Lotus for Calibnet
+
+   ```shell
+   make clean calibnet # Calibration with min 32GiB sectors
````
This comment isn't necessary.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+   git checkout <vX.X.X> # tag for a release, currently v1.15.0
+   ```
+
+2. Build and install Lotus for Calibnet
````
Calibnet should likely be in lowercase.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+   sudo make install
+   ```
+
+3. Fast sync using latest snapshot (still need several hours to fully sync the chain)
````
Don't use brackets (). Move this part of the sentence somewhere else.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+   lotus daemon --import-snapshot <path-to-calibration-snapshot>
+   ```
+
+4. [Create a wallet](https://lotu.sh/lotus/manage/manage-fil/#creating-a-wallet) and request some TestNet FIL from [Faucet](https://faucet.calibration.fildev.network/)
````
Why are you using `lotu.sh` links?

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+
+4. [Create a wallet](https://lotu.sh/lotus/manage/manage-fil/#creating-a-wallet) and request some TestNet FIL from [Faucet](https://faucet.calibration.fildev.network/)
+
+4. [Enable remote API access to your Lotus node](https://lotu.sh/developers/api-access/)
```
Again, don't use `lotu.sh` links. Use `docs.lotus.filecoin`

longfeiWan9 replied:
I will update all the lotu.sh.
This PR was created before we started using `docs.lotus.filecoin`

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+
+4. [Enable remote API access to your Lotus node](https://lotu.sh/developers/api-access/)
+
+If everything set up correctly, you should be able to verify to:
+
+- make sure that everything is working correctly, check the status of your local node:
+
+  ```shell with-output
+  #Check if the node sync to the highest tipset
+  lotus sync wait
+  ```
+
+  ```shell
+  Worker: 10161; Base: 778820; Target: 778820 (diff: 0)
+  State: complete; Current Epoch: 778820; Todo: 0
+  
+  Done!
+  ```
+
+- Check your wallet balance
+
+  ```shell with-output
+  #Check wallet balance
+  lotus wallet list 
+  ```
+
+  ```shell
+  Address                                    Balance        Nonce  
+  t154xvuihhicgluafwmohwzwmtmqp44pwwgewyvma  100 FIL          0
+  ```
````
This section shouldn't be split into bullet points. Write something like:

```
To ensure everything has been set up correct, check the status of your node, and then check the balance of you wallet.
```

Then show them how to do both those things. Don't chop it up into bullet points.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+  t154xvuihhicgluafwmohwzwmtmqp44pwwgewyvma  100 FIL          0
+  ```
+
+### Create a node.js project
````
Node.js should be capitalized.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+
+### Create a node.js project
+
+Let's also set up a Node.js project with necessary dependencies, we will write basic javascript functions to focus more on learning the code to interact with Filecoin Network. 
```
```suggestion
```

This sentence is redundant.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+
+Let's also set up a Node.js project with necessary dependencies, we will write basic javascript functions to focus more on learning the code to interact with Filecoin Network. 
+
+1. Create a new node.js project
```
```suggestion
1. Create a new Node.js project:
```

Capitalize Node.js. End list items with colons.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+
+   ```shell
+   mkdir build-with-lotus && cd build-with-lotus
+   npm init #Initialize an project
````
```suggestion
   npm init
```

Try not to comment your code. If something need explaining, explain it in a regular paragraph.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+   ```shell
+   npm install filecoin.js
+   ```
+
+
````
Make sure there's only one space at the end of sections. This was likely just a typo.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+
+Now you're ready to write code to connect to your Lotus node and interact with the Lotus APIs. Let's write a simple script to query the Lotus version to verify if everything works as expected.
+
+1. Create a `chainDataQuery.mjs` file in your project and add the following code. We use the `mjs` file format because it supports import statements.
```
End items with a colon.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+
+2. Run the following command in your project directory:
+
+    ```shell with-output
+    ~/build-with-lotus $ node basicQuery.mjs
+    ```
+
+    ```shell
+    {
+      Version: '1.15.0+calibnet+git.0ac1bbc7a',
+      APIVersion: 66816,
+      BlockDelay: 30
+    }
+    ```
````
The `with-output` class isn't enabled on this site. Instead, write something like:

````
```shell
cat text.txt
```

This should output something like:

```shelll
Hello world!
```
````

longfeiWan9 replied:
This will add a lot more unnecessary text in the tutorial. It is quite obvious that those blocks are output so that I will leave it as it is now.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+    }
+    ```
+
+If everything is set up correctly, your code will be able to connect to the Lotus node and return the chain data as shown above. Now, you're ready to explore more API features.
````
What if stuff doesn't work? Where should the reader look for more information?

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+- `LotusClient.client` - methods to interact with the storage and retrieval markets as a client
+- `LotusClient.paych` - methods to interact with and manage payment channels
+- `LotusClient.state` - methods to query, inspect, and interact with chain state
+- more ...
```
```suggestion
```

This isn't important.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
```diff
@@ -0,0 +1,646 @@
+- `LotusClient.state` - methods to query, inspect, and interact with chain state
+- more ...
+
+In the previous step, we already created a `chainDataQuery.mjs` file to demonstrate the basic steps to connect to the Lotus node and initialize a LotusClient to query chain data. Now let's write more code in there to learn other query functions that do not require auth-token to interact with the Lotus node.
```
```suggestion
In the previous step, we created a `chainDataQuery.mjs` file to demonstrate the basic steps to connect to the Lotus node and initialize a LotusClient to query chain data. Now, let's write more code in there to learn other query functions that do not require auth-token to interact with the Lotus node.
```

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+
+In the previous step, we already created a `chainDataQuery.mjs` file to demonstrate the basic steps to connect to the Lotus node and initialize a LotusClient to query chain data. Now let's write more code in there to learn other query functions that do not require auth-token to interact with the Lotus node.
+
+1. Get current chain head
+
+   This function will return the current head of the chain, including current block height, a list of blocks, and a list of CIDs.
+     ```javascript with-output
+      //Query the current block head
+      const chainHead = await lotusClient.chain.getHead();
+      console.log(chainHead);
+     ```
+
+      ```shell
+      {
+        Cids: [
+          {'/': 'bafy2bzacecdkonmhngylnnhrk4azkg2wkgcm6cnm5qn5sk4ww5cszjlvkgkd6'},
+          {'/': 'bafy2bzaceaglcpzhd5gfrzdyt7ce3e5asnbfz3s3stbqyxniziny5snewbpbg'}
+          ],
+        Blocks: [
+          {...},
+          {...}
+          ],
+        Height: 781268
+      }
+      ```
````
There's a lot of lines here for a single list item. Should this section be split into smaller headers maybe? Or just remove the `This function will return [...]` bit and tell the user what they need to do. If you need to explain what's going on in the code, do it after you've told them what the code should look like.

longfeiWan9 replied:
If I fully post the chain head body here, it will be so many lines (might be 50+ for each block). So I just `...` as an ellipsis for block body. Do you have any suggestions that I can use instead of `...` or it is acceptable?

longfeiWan9 replied:
Fixed it.  Thank you, Mosh

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,646 @@
+      }
+      ```
+
+2. Get messages in a block
+
+    ```javascript with-output
+    //Query all the messages in a block
+    const tipSet = await lotusClient.chain.getTipSetByHeight(781267);
+    const messages = await lotusClient.chain.getBlockMessages(tipSet.Cids[0]);
+    console.log(messages);
+    ```
+
+    ```shell
+    {
+      BlsMessages: [
+        {
+          Version: 0,
+          To: 't3ukhj6tpkgxjknu54opiaej2vrjz7nh7gzodkqlhpfphc6gxkogzmojv2cnlpgcuwkvnyhloctnc6lmlvceuq',
+          From: 't3ws4jhb7g4n5s7t3mwsc2enzfija2kzo3p7vkqdqbiaqrabqhmfxzzy3knrvn3ykp75okab6pse2nxud3ineq',
+          ...
+        }
+      ],
+      SecpkMessages: [ { Message: [Object], Signature: [Object], CID: [Object] } ],
+      Cids: [
+        {'/': 'bafy2bzaceapv6ms4m3x4gdaefmpilgsqfhfsrvrciw4iux4n7va47es4vudpa'},
+        {'/': 'bafy2bzaceaxaz5ews6a2jnwmeafhi2p5xpwjz7kogilakblp3q3wpxa3ax2xy'}
+      ]
+    }
+    ```
+
+3. Get wallet balance
+
+    With a given TestNet wallet address, you can query its balance by calling lotusClient.wallet.balance() function.
+
+    ```javascript
+    //Query Wallet balance
+    const walletBalance = await lotusClient.wallet.balance("t1ne72cbn6r55wea7ifjv4ypyti7t2df5dumsjhzq");
+    ```
````
See previous comment about explaining your code _after_ you've shown it to the user.

longfeiWan9 replied:
Fix it.

longfeiWan9 replied:
Fix it.

johnnymatthews reviewed (changes requested):
This is a really, _really_ long tutorial, and there's quite a bit of work that needs to be done to fix this tutorial up. I've got about half way through the tutorial with comments, and a lot of my requests are duplicated.

I think we should consider chopping this tutorial up into several smaller pages.

mishmosh commented:
I just pushed a commit directly to this branch. It contains many small fixes:
- grammar, typos, & sentence cleanups
- replaced the lotu.sh links with lotus.filecoin.io
- clarified the information on hosted nodes vs. local nodes after chatting with @longfeiWan9 on specifics (addresses your comments @johnnymatthews, but also clarifies the section overall)

Agree that it’s long, but I think this actually hangs together and there's no section that can be easily split off from the others. Instead, I split it into 4 numbered sections and created a linked table of contents. 

I believe this addresses all of @johnnymatthews' comments, except for the general notes about putting explanations after code.

This now looks good to me, but @johnnymatthews let us know if the code-before-explanation thing is a blocker.

mishmosh commented:
I just pushed a commit directly to this branch. It contains many small fixes:
- grammar, typos, & sentence cleanups
- replaced the lotu.sh links with lotus.filecoin.io
- clarified the information on hosted nodes vs. local nodes after chatting with @longfeiWan9 on specifics (addresses your comments @johnnymatthews, but also clarifies the section overall)

Agree that it’s long, but I think this actually hangs together and there's no section that can be easily split off from the others. Instead, I split it into 4 numbered sections and created a linked table of contents. 

I believe this addresses all of @johnnymatthews' comments, except for the general notes about putting explanations after code.

This now looks good to me, but @johnnymatthews let us know if the code-before-explanation thing is a blocker.

mishmosh commented:
Also, you’ll both hate me but after getting a fresh look at the nearly-finished tutorial, I think this belongs in the Lotus docs. From the Filecoin docs, we can link to it from the Build section.

longfeiWan9 commented:
@mishmosh  Thanks for all the fixes. I will take one more look and add a few fixes according to @johnnymatthews comments. 

Initially, we were planning to add it to Lotus docs as well. But I feel like this is more of guiding developers to build on top of Filecoin instead of using Lotus. So I end up putting it in Filecoin docs. But still, I am not very clear on how to decide what goes to which docs. If you both think it should go to Lotus docs, let's just move it over once we finished reviewing. :)

johnnymatthews commented:
I went through the whole guide and made some changes directly. Here are my notes:

1. Stick to a perspective. You started out saying "we will do ..." but then switched to "next you should ...", and further on in the guide, you also used things like "then I ran ...". In tutorials like this, it's best to use sentences like "let's install Lotus" and "next we're going to". 
1. When writing a numbered list, always start each item with `1.`. When the markdown is converted into HTML, the parser will count the number of `1.`s and changes the number where necessary. If you instead use actual numbers in your markdown, the parser will use those numbers explicitly. So if your numbers are out of order, or you missed one, the parser won't realize, and it'll display incorrectly in the HTML. 
1. I'm not a huge fan of linking off to other tutorials for things like creating a wallet. I get that we don't want to repeat information, but it's not great UX to have the reader bounce around tutorials and pages. If we're concerned about this tutorial being too long by including this kind of stuff, we should chop this tutorial into separate pages.

Here are some changes that we need to make:

1. How does a user backup their `encryptedWallet` object?
1. When adding data to the Lotus node, should we tell the reader to move the file/s into the same project folder as the script. That might make it easier to reference the path in the `importResults` object.
1. Can we get an example of the output from the `importResults` object creation?

johnnymatthews reviewed (changes requested):
1. How does a user backup their `encryptedWallet` object?
1. When adding data to the Lotus node, should we tell the reader to move the file/s into the same project folder as the script. That might make it easier to reference the path in the `importResults` object.
1. Can we get an example of the output from the `importResults` object creation?

longfeiWan9 commented:
> 1. How does a user backup their `encryptedWallet` object?

Added a line to explain how to back `encryptedWallet`.

> 2. When adding data to the Lotus node, should we tell the reader to move the file/s into the same project folder as the script? That might make it easier to reference the path in the `importResults` object.

Since the js project is able to read files from folders outside the project, so I think it is not necessary to add one more step. we can keep it as it is now.

> 3. Can we get an example of the output from the `importResults` object creation?

Added.
@johnnymatthews can you take another look? Thank you.

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,664 @@
+    const tipSet = await lotusClient.chain.getTipSetByHeight(781267);
+    const messages = await lotusClient.chain.getBlockMessages(tipSet.Cids[0]);
+    console.log(messages);
+    ```
+    Using the above code in`chainDataQuery.mjs` , we can also retrieve messages from a certain block height. And the returned message body will look like the following .
+    ```shell
````
````suggestion

    ```
    Using the above code in`chainDataQuery.mjs`, we can also retrieve messages from a certain block height. And the returned message body will look like the following.

    ```shell
````

johnnymatthews commented on content/en/build/building-with-lotus-APIs.md:
````diff
@@ -0,0 +1,664 @@
+    This will output your auth token. Save this somewhere, we'll be using it later:
+
+    ```shell
+    eyJhbGciOiJIUzI1NiIsInRaaaaaa.xxxxxxx.bbbbbbbbbbbvq1W0ZjqrXHygd6fBRk
````
```suggestion
eyJhbGciOiJIUzI1NiIsInRaaaaaa.xxxxxxx.bbbbbbbbbbbvq1W0ZjqrXHygd6fBRk
```
